be/src/exec/workgroup/work_group.cpp

```diff
@@ -265,6 +265,34 @@ StatusOr<PriorityThreadPool::Task> IoWorkGroupQueue::pick_next_task() {
     return wg->pick_io_task();
 }
 
+StatusOr<PriorityThreadPool::Task> IoWorkGroupQueue::pick_next_real_time_task() {
+    std::unique_lock<std::mutex> lock(_realtime_io_mutex);
+
+    if (_is_closed) {
+        return Status::Cancelled("Shutdown");
+    }
+
+    while (_real_time_wg->io_task_queue_size() == 0) {
+        _realtime_io_cv.wait(lock);
+        if (_is_closed) {
+            return Status::Cancelled("Shutdown");
+        }
+    }
+
+    return _real_time_wg->pick_io_task();
+}
+
+StatusOr<PriorityThreadPool::Task> WorkGroupManager::pick_next_task_for_io(bool is_real_time_type) {
+    if (is_real_time_type) {
+        return _wg_io_queue.pick_next_real_time_task();
+    }
+    return _wg_io_queue.pick_next_task();
```

Comment:
If we implement two different io `WorkGroupQueue` for real-time and non-real-time, and make `io_dispatcher` own `WorkGroupQueue`, then maybe we needn't add method `pick_next_real_time_task` and `pick_next_ask`, and needn't judge whether is real-time here.

Reply:
+1